eis_toolkit/spatial_analyses/cell_based_association.py

```diff
@@ -0,0 +1,573 @@
+    assert len(subset_target_attribute_values) != 0
+
+    return dummification, column, list(identified_values)
+
+
+@beartype
+def _get_disc(geodata: gpd.GeoDataFrame, grid: gpd.GeoDataFrame, buff_radius: Union[int, float]) -> gpd.GeoDataFrame:
+    """Intermediate utility.
+
+    Defines disks with radii defined from the centroids of the cells of a grid.
+
+    Args:
+        geodata: Geodata to intersect with the grid.
+        grid: Grid from which the discs will be created.
+        buff_radius: Radius for created buffer discs
+
+    Returns:
+        GeoDataFrame containing the records from the grid buffers/geodata
+        intersection.
+    """
+
+    # Deletion of cells that do not intersect shapefile geometries
+    cells_w_bound = gpd.sjoin(grid, geodata, how="inner", predicate="intersects")
+    cells_w_bound["index"] = cells_w_bound.index
+    cells_w_bound = cells_w_bound.dissolve(by="index")
+    drop_list = geodata.columns
+    drop_list = drop_list.drop("geometry")
+    cells_w_bound = cells_w_bound.drop(drop_list, axis="columns")
+
+    # Creation of the centroids of the mesh
+    centX = cells_w_bound.centroid.x
+    centY = cells_w_bound.centroid.y
+    df = pd.DataFrame({"CentX": centX, "CentY": centY})
+    points = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.CentX, df.CentY), crs=geodata.crs)
+
+    # Transformation into a disc with a given radius
+    disc = gpd.GeoDataFrame.copy(points)
+    disc["geometry"] = disc.geometry.buffer(buff_radius, resolution=16)
+
+    return gpd.GeoDataFrame(disc, geometry="geometry", crs=geodata.crs)
+
+
+@beartype
+def _prepare_disc(
+    geodata: gpd.GeoDataFrame,
+    disc: gpd.GeoDataFrame,
+    column: str,
+    subset_target_attribute_values: list,
+) -> Tuple[pd.Series, gpd.GeoDataFrame]:
+    """Intermediate utility.
+
+    Compute spatial join and generate the table of presence of the different
+    attributes.
+
+    Args:
+        geodata: The geodata object corresponding to the vector file.
+        disc: Existing discs 'paving'.
+        column: Target column to extract from the vector file.
+        subset_target_attribute_values: Subset of the relevant values
+            of the target attribute
+
+    Returns: A boolean indicating if the target attribute must be dummified
+        and an intersection grid between the mesh and the vector file
+    """
+    dummification, target, identified_values = _check_and_prepare_param(geodata, column, subset_target_attribute_values)
+
+    # Spatial join of the intersection between the grid and the current map
+    join_grid = gpd.sjoin(disc, geodata, how="inner", predicate="intersects")
+
+    if dummification:
+        tmp = pd.get_dummies(join_grid[join_grid[target].isin(identified_values)][[target]], prefix="", prefix_sep="")
+        indicators = tmp.groupby(tmp.index).max()
+    else:
+        tmp = join_grid[target]
+        indicators = tmp.groupby(tmp.index).max()
+    return indicators, join_grid
```

Comment:
_prepare_disc and _prepare_grid seem to be the same function, so you could remove one of them.